vllm/core/scheduler.py

```diff
@@ -1584,64 +1636,174 @@ def _get_num_lookahead_slots(self, is_prefill: bool,
 
         return self.scheduler_config.num_lookahead_slots
 
-    def _get_num_new_tokens(self, seq_group: SequenceGroup,
-                            status: SequenceStatus, enable_chunking: bool,
-                            budget: SchedulingBudget) -> int:
-        """Get the next new tokens to compute for a given sequence group
-            that's in a given `status`.
+    def _get_num_new_tokens_to_schedule(
```

Comment:
It's a bit hard to tell the difference in terms of functionalities between this function and `_get_num_new_tokens_uncached_and_cached` from their names. For example, the term `to_schedule` doesn't convey its functionality but the purpose. Could we try to figure out a better naming for both of them?

Reply:
Agree - 

Maybe:
- (this) `_get_num_new_tokens_to_schedule` -> `_get_num_new_uncached_and_cached_tokens`
- (the routine below) -> `_get_num_new_uncached_and_cached_tokens_for_seq` ?

Reply:
So the current routine below is just used by this one? In this case can we just rename `_get_num_new_tokens_to_schedule` -> `_get_num_new_uncached_and_cached_tokens` and move the routine below to be the inner function?